### PR TITLE
fix(Switch/native): moves defaultOn to a separate useEffect

### PR DESCRIPTION
**Pull request by shubhnik**

### CODE REVIEW

File: src/atoms/Switch/Switch.native.js [hunk 1]
```diff
@@ -189,6 +189,20 @@ const Switch = ({ disabled, on, defaultOn, onChange, size, testID }) => {
     new Animated.Value(animationConfig.off.rightSpace),
   );
 
+  useEffect(() => {
+    if (defaultOn) {
```
Comment (kamleshchandnani) [inline src/atoms/Switch/Switch.native.js:193]: why was this required?

Comment (shubhnik) [reply src/atoms/Switch/Switch.native.js:193]: This was for `defaultOn`. You did the eslint formating thing, which introduced the bug as I hadn't added all the dependencies in the useEffect.
